setup-scripts/setup_host_rpi.sh

```diff
@@ -66,20 +42,45 @@ function append_if_missing() {
 }
 
 function forward_port() {
-    # usage: forward_port <host port> <superbird port>
-    # forward a tcp port to access service on superbird via host
-    #   if no superbird port is provided, same port number is used for both
+    # Usage: forward_port <host port> <superbird port>
+    # Forwards a tcp port to access service on superbird via host.
+    # If no superbird port is provided, same port number is used for both.
     SOURCE="$1"
     DEST="$2"
     if [ -z "$DEST" ]; then
         DEST="$SOURCE"
     fi
-    iptables -t nat -A PREROUTING -p tcp -i eth0 --dport "$SOURCE" -j DNAT --to-destination "${USBNET_PREFIX}.2:$DEST"
-    iptables -t nat -A PREROUTING -p tcp -i wlan0 --dport "$SOURCE" -j DNAT --to-destination "${USBNET_PREFIX}.2:$DEST"
+    iptables -t nat -A PREROUTING -p tcp -i "${WAN_INTERFACE}" --dport "$SOURCE" -j DNAT --to-destination "${USBNET_PREFIX}.2:$DEST"
     iptables -A FORWARD -p tcp -d "${USBNET_PREFIX}.2" --dport "$DEST" -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
 }
 
-# detect if car thing is plugged in
+# PRE_FLIGHT_CHECKS
+
+# Need to be root
+if [ "$(id -u)" != "0" ]; then
+    echo "Must be run as root"
+    exit 1
+fi
+
+# Does not run on BSD
+if [ "$(uname -s)" != "Linux" ]; then
+    echo "Only works on Linux!"
+    exit 1
+fi
+
+# Does not run on anything but Raspbian Bullseye
+if ! lsb_release -a | grep -q "Raspbian GNU/Linux 11 (bullseye)"; then
+    echo "Current OS not supported!"
+    exit 1
+fi
+
+# Detect if usb0 CarThing NIC is already configured.
+if ! ip addr show "${CT_INTERFACE}" | grep -q "192.168.1 "; then
+    echo "No inactive network interface found. This may occur if the script was already run, or if your Spotify Car Thing is not plugged in."
+    exit 1
+fi
```

Comment:
Yes this works, the problem is that before you were checking if the string "inet " was present. However because of APIPA it was getting an automatically configured IP address. Thus failing the check. This checks to see if it has been explicitly configured for the Car Thing subnet.